runtime/server/catalog.go

```diff
@@ -33,6 +35,29 @@ func (s *Server) ListCatalogEntries(ctx context.Context, req *runtimev1.ListCata
 		return nil, status.Error(codes.Unknown, err.Error())
 	}
 
+	if req.Type == runtimev1.ObjectType_OBJECT_TYPE_METRICS_VIEW {
+		// apply access policies to metrics views
+		accessibleEntries := make([]*drivers.CatalogEntry, 0)
+
+		for _, entry := range entries {
+			policy, err := s.runtime.ResolveMetricsViewPolicy(auth.GetClaims(ctx).Attributes(), req.InstanceId, entry.GetMetricsView(), entry.UpdatedOn)
+			if err != nil {
+				return nil, err
+			}
+
+			if policy != nil && !policy.HasAccess {
+				continue
+			}
+			// deep copy the catalog entry, so we can filter dimensions which mutates the metricsview
+			newEntry := deepcopy.Copy(entry).(*drivers.CatalogEntry)
+			mv := newEntry.GetMetricsView()
+
+			filterDimensions(policy, mv)
+			accessibleEntries = append(accessibleEntries, newEntry)
```

Comment:
It would be preferable only to clone the metrics view if it has dimension filters (most will probably only have row filters)